services/users/src/routes/company.ts

```diff
@@ -59,52 +56,46 @@ companyRoutes.route("/:id").put(
 
 companyRoutes.route("/:id/employees/add").post(
   asyncHandler(async (req, res) => {
-    const {id} = req.params;
+    const { id } = req.params;
     const company = await CompanyModel.findById(id);
 
     if (!company) {
-      res.status(400);
       throw new BadRequestError("Company not found");
     }
 
-    const emails = req.body.split(",");
+    const emails = req.body.employees.split(",");
     const newEmployees: string[] = [];
 
-    emails.forEach(async (index: number) => {
-      const email = emails[index];
-      const getUser = await getAuth().getUserByEmail(email);
-      const getUid = getUser.uid;
-      newEmployees.push(getUid);
-    });
-    // }) {
-    //     var email = emails[i];
-    //     var getUser = getAuth().getUserByEmail(email);
-    //     var getUid = (await getUser).uid
-    //     newEmployees.push(getUid);
-    // }
-
-    const addEmployees = await CompanyModel.findByIdAndUpdate(id, {
-      employees: newEmployees,
+    emails.forEach(async (email: string) => {
+      const user = await getAuth().getUserByEmail(email);
+      newEmployees.push(user.uid);
     });
 
-    return res.status(200).send(addEmployees);
+    const currEmployees = await CompanyModel.findByIdAndUpdate(
+      id,
+      {
+        employees: company.employees.concat(newEmployees),
+      },
+      { new: true }
+    );
```

Comment:
So we can't just concat the new employees, we have to make sure the same user isn't duplicated. We don't want the same user added twice, so can you do some filtering to make sure each employee uid only appears once?

Reply:
Sorry about that, didn't think about that when implementing. Should be fixed now